src/engine/loopingcontrol.h

```diff
@@ -32,14 +32,14 @@ class LoopingControl : public EngineControl {
     // process() updates the internal state of the LoopingControl to reflect the
     // correct current sample. If a loop should be taken LoopingControl returns
     // the sample that should be seeked to. Otherwise it returns currentSample.
-    double process(const double dRate,
+    virtual double process(const double dRate,
```

Comment:
Yes, correct.  I think the right way to do this is to make RAMAN take a regular EngineControl*, and that can either be this looping control or the mocked looping control.  The looping stub should not inherit from LoopingControl, because it might accidentally inherit some actual loopingcontrol behavior, and then it's not a proper stub.